File: csharp/2007/mr-brooks-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "Mr. Brooks" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "Mr. Brooks",
                year: 2007,
                plot: "A psychological thriller about a man who is sometimes controlled by his murder-and-mayhem-loving alter ego.",
                rating: 7.3
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "Mr. Brooks",
                year: 2007
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}

File: python/2007/mr-brooks-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add Mr. Brooks to the database
movies.insert(
    title="Mr. Brooks", 
    year=2007, 
    plot="A psychological thriller about a man who is sometimes controlled by his murder-and-mayhem-loving alter ego.", 
    rating=7.3
)

# Confirm that the movie was added
movie = movies.select(
    title="Mr. Brooks", 
    year=2007
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    